src/main.c

```diff
@@ -324,13 +324,30 @@ int main(int argc, char **argv)
     else
         init_dos(argc - 1, argv + 1);
 
+#if defined(_AIX) || defined(__sun)
+    struct sigaction timer_action, exit_action;
+    exit_action.sa_handler = exit_handler;
+    timer_action.sa_handler = timer_alarm;
+    sigemptyset (&exit_action.sa_mask);
+    sigemptyset (&timer_action.sa_mask);
+    exit_action.sa_flags = timer_action.sa_flags = 0;
+
+    sigaction (SIGALRM, &timer_action, NULL);
+    sigaction (SIGHUP, &exit_action, NULL);
+    sigaction (SIGHUP, &exit_action, NULL);
+    sigaction (SIGINT, &exit_action, NULL);
+    sigaction (SIGQUIT, &exit_action, NULL);
+    sigaction (SIGPIPE, &exit_action, NULL);
+    sigaction (SIGTERM, &exit_action, NULL);
+#else
```

Comment:
Sorry, I think it is better to simply use sigaction in all cases. Is there any platform that does not support sigaction?

Reply:
I’ll verify tonight.

Reply:
Testing now